### PR TITLE
Fix post-renderer flag not working in diff and template subcommand

### PR DESCRIPTION
**Pull request by xiaomudk**
https://github.com/helmfile/helmfile/issues/665
Signed-off-by: xiaomudk <xiaomudk@gmail.com>

### CODE REVIEW

File: test/integration/test-cases/postrender.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+postrender_diff_tmp=$(mktemp -d)
+postrender_diff_reverse=${postrender_diff_tmp}/postrender.diff.build.yaml
+postrender_template_reverse=${postrender_diff_tmp}/postrender.template.build.yaml
+
+test_start "postrender diff"
+info "Comparing postrender diff output ${postrender_diff_reverse} with ${postrender_case_output_dir}/result.yaml"
+for i in $(seq 10); do
+    info "Comparing build/postrender-diff #$i"
+    ${helmfile} -f ${postrender_case_input_dir}/${config_file} diff --concurrency 1 --post-renderer ./add-cm1.bash &> ${postrender_diff_reverse} || fail "\"helmfile diff\" shouldn't fail"
```
Comment (yxxhero) [inline test/integration/test-cases/postrender.sh:30]: @xiaomudk I can't find any code about `add-cm2.bash`.  I would like to use `add-cm2.bash` in the template stage. WDYT?

File: test/integration/test-cases/postrender.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    diff -u  ${postrender_diff_out_file} ${postrender_diff_reverse} || fail "\"helmfile diff\" should be consistent"
+    echo code=$?
+done
+test_pass "postrender diff"
+
+test_start "postrender template"
+info "Comparing postrender template output ${postrender_template_reverse} with ${postrender_case_output_dir}/result.yaml"
+for i in $(seq 10); do
+    info "Comparing build/postrender-diff #$i"
+    ${helmfile} -f ${postrender_case_input_dir}/${config_file} template --concurrency 1 --post-renderer ./add-cm1.bash &> ${postrender_template_reverse} || fail "\"helmfile template\" shouldn't fail"
```
Comment (yxxhero) [inline test/integration/test-cases/postrender.sh:40]: here.

File: test/integration/test-cases/postrender.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    ${helmfile} -f ${postrender_case_input_dir}/${config_file} diff --concurrency 1 --post-renderer ./add-cm1.bash &> ${postrender_diff_reverse} || fail "\"helmfile diff\" shouldn't fail"
+    diff -u  ${postrender_diff_out_file} ${postrender_diff_reverse} || fail "\"helmfile diff\" should be consistent"
+    echo code=$?
+done
+test_pass "postrender diff"
+
+test_start "postrender template"
+info "Comparing postrender template output ${postrender_template_reverse} with ${postrender_case_output_dir}/result.yaml"
+for i in $(seq 10); do
+    info "Comparing build/postrender-diff #$i"
```
Comment (yxxhero) [inline test/integration/test-cases/postrender.sh:39]:
```suggestion
    info "Comparing build/postrender-template #$i"
```

File: test/integration/test-cases/postrender.sh (hunk 1 above)
Comment (xiaomudk) [reply test/integration/test-cases/postrender.sh:30]: in `test/integration/test-cases/postrender/input/helmfile.yaml`
```
    dep:
      templates:
      - |
        apiVersion: v1
        kind: ConfigMap
        metadata:
          name: {{`{{ .Release.Name }}`}}-2
          namespace: {{`{{ .Release.Namespace }}`}}
        data:
          bar: BAR
  postRenderer: ./add-cm2.bash
- name: baz
  chart: ../../../charts/raw
```

Comment (yxxhero) [reply test/integration/test-cases/postrender.sh:30]: @xiaomudk but I don't see any output about add-cm2. maybe the --post-renderer overwrites that release level.

**Comment by xiaomudk** [comment]
@yxxhero  Priorities are as follows:
```
func (st *HelmState) appendPostRenderFlags(flags []string, release *ReleaseSpec, helm helmexec.Interface) []string {
	switch {
	// helm.GetPostRenderer() comes from cmd flag.
	case helm.GetPostRenderer() != "":
		flags = append(flags, "--post-renderer", helm.GetPostRenderer())
	case release.PostRenderer != nil && *release.PostRenderer != "":
		flags = append(flags, "--post-renderer", *release.PostRenderer)
	case st.HelmDefaults.PostRenderer != nil && *st.HelmDefaults.PostRenderer != "":
		flags = append(flags, "--post-renderer", *st.HelmDefaults.PostRenderer)
	}
	return flags
}
```

I think this is problematic, `release` should have the highest priority

**Comment by yxxhero** [comment]
@xiaomudk I think so. if `release` has the highest priority. cmd flag can overwrite `helmDefault`. it will be more flexible.


@mumoshu WDYT?

**Comment by yxxhero** [comment]
@xiaomudk just do it.

**Review by yxxhero — APPROVED** [review]

**Pull request merged**